pytorch_lightning/core/optimizer.py

```diff
@@ -337,15 +337,27 @@ def _get_default_scheduler_config() -> Dict[str, Any]:
         "reduce_on_plateau": False,  # most often not ReduceLROnPlateau scheduler
         "monitor": None,  # value to monitor for ReduceLROnPlateau
         "strict": True,  # enforce that the monitor exists for ReduceLROnPlateau
-        "opt_idx": None,  # necessary to store opt_idx when optimizer frequencies are specified
+        "opt_idx": None,  # opt_idx assigned internally if not assigned by user
     }
 
 
-def _validate_scheduler_optimizer(optimizers: List[Any], lr_schedulers: List[Any]) -> None:
-    if any(sch["scheduler"].optimizer not in optimizers for sch in lr_schedulers):
-        raise MisconfigurationException(
-            "Some schedulers are attached with an optimizer that wasn't returned from `configure_optimizers`."
-        )
+def _set_scheduler_opt_idx(optimizers: List[Any], lr_schedulers: List[Any]) -> None:
+    for sch in lr_schedulers:
+
+        for opt_idx, opt in enumerate(optimizers):
+            if sch["scheduler"].optimizer is opt:
+                if sch["opt_idx"] is not None and sch["opt_idx"] != opt_idx:
+                    raise MisconfigurationException(
+                        "`opt_idx` set inside scheduler config does not match with index"
+                        " of respective optimizer returned from `configure_optimizers`."
```

Comment:
Isn't it clearer with "the"s?

```suggestion
                        "`opt_idx` set inside scheduler config does not match with the index"
                        " of the respective optimizer returned from `configure_optimizers`."
```